pkg/registry/apis/provisioning/jobs/pullrequest.go

```diff
@@ -0,0 +1,242 @@
+package jobs
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"log/slog"
+	"net/url"
+	"path"
+
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/repository"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
+)
+
+// resourcePreview represents a resource that has changed in a pull request.
+type resourcePreview struct {
+	Filename             string
+	Path                 string
+	Action               string
+	Type                 string
```

Comment:
```suggestion
	Kind                 string
```
yes pedantic, but can we use the Kind from GVK -- eg "Dashboard"